src/sql/workbench/api/node/mainThreadNotebookDocumentsAndEditors.ts

```diff
@@ -361,26 +361,9 @@ export class MainThreadNotebookDocumentsAndEditors extends Disposable implements
 			pinned: !options.preview
 		};
 		let trusted = uri.scheme === Schemas.untitled;
-		let model = new NotebookInputModel(uri, undefined, trusted, undefined, undefined, undefined, options.connectionId);
-		let providerId = options.providerId;
-		let providers: string[] = undefined;
-		// Ensure there is always a sensible provider ID for this file type
-		providers = getProvidersForFileName(uri.fsPath, this._notebookService);
-		// Try to use a non-builtin provider first
-		if (providers) {
-			providerId = providers.find(p => p !== DEFAULT_NOTEBOOK_PROVIDER);
-			if (!providerId) {
-				providerId = model.providerId;
-			}
-		}
-		model.providers = providers;
-		model.providerId = providerId;
-		model.defaultKernel = options && options.defaultKernel;
-		model.providers.forEach(provider => {
-			let standardKernels = getStandardKernelsForProvider(provider, this._notebookService);
-			model.standardKernels = standardKernels;
-		});
-		let input = this._instantiationService.createInstance(NotebookInput, undefined, model);
+		let input = this._instantiationService.createInstance(NotebookInput, uri.fsPath, uri);
+		input.isTrusted = trusted;
+		input.defaultKernel = options && options.defaultKernel;
```

Comment:
Please assign connectionProfileId here, seems like the point at which we're missing it. Or pass to the NotebookInput constructor as an optional parameter